### DISCUSSION TITLE
Support of syncwaves with application sets

### DISCUSSION THREAD

**Discussion by shamonshan** [argoproj/argo-cd#16843] [Q&A] [answered]
Currently i  am managing applications using application sets , but the problem the sync waves not works with the  generated application.

My requirement is something like this

Deploy the applications in sequential order such as 
Application A -> Application B -> Application C

Is there any workaround to use sync waves with application set

**Comment by iflan7744** [comment, marked as answer]
Try progressive sync
https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/

**Reply by shamonshan** [reply]
Thanks one more question how does the application order is behave if more than one applicationset?
Eg: 
   1. Application A and Application B in -> Applicationset1
   2. Application C in ->Applicationset2

**Reply by FernandoMiguel** [reply]
each applicationset follows their own progressive sync

**Reply by iflan7744** [reply]
@shamonshan did you find it useful?

**Reply by shamonshan** [reply]
@iflan7744  Thanks   "each applicationset follows its own progressive sync" this mean the progressive sync between applicationsets is independent??

**Reply by iflan7744** [reply]
**Yes**. Make it **Answered** so it will be helpful for others.
